### PR TITLE
Moonpay ask for camera permissions, if not show a modal to change on settings

### PR DESCRIPTION
**Pull request by amanalili**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android

### CODE REVIEW

File: src/components/services/PermissionsManager.js [hunk 1]
```diff
@@ -107,6 +110,25 @@ export async function requestPermission(data: Permission): Promise<PermissionSta
   return status
 }
 
+export async function enablePermissionOnSettings(data: Permission, name: string): Promise<void> {
+  const status: PermissionStatus = await check(PERMISSIONS[OS][PERMISSIONS_ITEM[data]])
+
+  if (status === RESULTS.DENIED) {
+    requestPermission(data)
```
Comment (amanalili) [inline src/components/services/PermissionsManager.js:124]: return requestPermission(data)

File: src/components/services/PermissionsManager.js [hunk 2]
```diff
@@ -107,6 +110,25 @@ export async function requestPermission(data: Permission): Promise<PermissionSta
...
+  } else if (status === RESULTS.BLOCKED) {
+    const answer = await Airship.show(bridge => (
```
Comment (amanalili) [inline src/components/services/PermissionsManager.js:126]: use .then and remove await

File: src/constants/plugins/GuiPlugins.js [hunk 3]
```diff
@@ -23,7 +23,8 @@ export const guiPlugins: { [pluginId: string]: GuiPlugin } = {
     baseQuery: { apiKey: 'pk_live_Y1vQHUgfppB4oMEZksB8DYNQAdA4sauy' },
     queryPromoCode: 'apiKey',
     displayName: 'MoonPay',
-    permissions: ['camera']
+    permissions: ['camera'],
+    settingsPermission: true
```
Comment (amanalili) [inline src/constants/plugins/GuiPlugins.js:27]: mandatoryPermissions

File: src/components/services/PermissionsManager.js [hunk 4]
```diff
@@ -111,6 +113,11 @@ export async function requestPermission(data: Permission): Promise<PermissionSta
 }
 
 export async function enablePermissionOnSettings(data: Permission, name: string): Promise<void> {
+  const permissionLimits = await AsyncStorage.getItem(SETTINGS_PERMISSION_LIMITS)
```
Comment (amanalili) [inline src/components/services/PermissionsManager.js:116]: Use disklet

File: src/components/services/PermissionsManager.js [hunk 5]
```diff
@@ -3,14 +3,19 @@
...
+import s from '../../locales/strings.js'
 import { type Permission, type PermissionsState, type PermissionStatus } from '../../reducers/PermissionsReducer.js'
 import { connect } from '../../types/reactRedux.js'
+import { ButtonsModal } from '../modals/ButtonsModal.js'
 import { type ContactsPermissionResult, ContactsPermissionModal } from '../modals/ContactsPermissionModal.js'
 import { Airship, showError } from './AirshipInstance.js'
 
 const IS_CONTACTS_PERMISSION_SHOWN_BEFORE = 'IS_CONTACTS_PERMISSION_SHOWN_BEFORE'
+const SETTINGS_PERMISSION_LIMITS = 'SETTINGS_PERMISSION_LIMIT'
+const SETTINGS_PERMISSION_QUANTITY = 3
```
Comment (amanalili) [inline src/components/services/PermissionsManager.js:18]: Create a new file constantSettings.js and put these constants

File: src/components/scenes/GuiPluginViewScene.js [hunk 6]
```diff
@@ -194,10 +195,17 @@ class GuiPluginView extends React.Component<Props, State> {
...
     const { plugin } = route.params
     const { permissions = [] } = plugin
-    for (const name of permissions) await requestPermission(name)
+    const { displayName, mandatoryPermissions } = plugin
+    const mandatory = mandatoryPermissions != null && mandatoryPermissions ? mandatoryPermissions : false
+    for (const permission of permissions) {
+      const deniedPermission = await enablePermissionOnSettings(state.core.disklet, permission, displayName, mandatory)
+      if (deniedPermission) return true
+    }
+
```
Comment (amanalili) [inline src/components/scenes/GuiPluginViewScene.js:207]: if (deniedPermission) return handlePluginBack()

File: src/components/services/PermissionsManager.js [hunk 7]
```diff
@@ -100,13 +103,54 @@ export async function requestPermission(data: Permission): Promise<PermissionSta
...
+  const permissionLimits = await disklet.getText(SETTINGS_PERMISSION_LIMITS).catch(_ => null) // Ignore and don't throw error when file not found
+  const permissionLimit = permissionLimits != null && JSON.parse(permissionLimits)[data] != null ? JSON.parse(permissionLimits)[data] : 0
+
+  // Check to ignore the permission checks if not mandatory and already past the limit
+  if (!mandatory && permissionLimit >= SETTINGS_PERMISSION_QUANTITY) return false
+
+  const fullPermision: string = PERMISSIONS[OS][PERMISSIONS_ITEM[data]]
+  const status: PermissionStatus = await check(fullPermision)
+
+  if (mandatory && status === RESULTS.UNAVAILABLE) return true
```
[the remark this answers is not in the document]
Comment (amanalili) [reply src/components/services/PermissionsManager.js:127]: Actually checking the code. This one is right. Check the in line comments for explanation

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**